pydantic_ai_slim/pydantic_ai/models/anthropic.py

```diff
@@ -247,12 +247,29 @@ async def _messages_create(
         tools = self._get_tools(model_request_parameters)
         tools += self._get_builtin_tools(model_request_parameters)
         tool_choice: BetaToolChoiceParam | None
+        thinking = model_settings.get('anthropic_thinking', NOT_GIVEN)
+        temperature = model_settings.get('temperature', NOT_GIVEN)
+        top_p = model_settings.get('top_p', NOT_GIVEN)
 
         if not tools:
             tool_choice = None
         else:
             if not model_request_parameters.allow_text_output:
                 tool_choice = {'type': 'any'}
+                if thinking and thinking.get('type') == 'enabled':
+                    tool_choice = {'type': 'auto'}
+                    if temperature and temperature != 1:
```

Comment:
I'd rather have the user see an error and adjust this manually, than to change it for them and log warnings. So please remove this from the PR for now and keep it focused on the tool choice issue. If others report the temperature/top_p issue and find it hard to work around, we can consider something like this then.